packages/core/src/browser/quick-open/quick-input-service.ts

```diff
@@ -133,6 +133,7 @@ export class QuickInputService {
         this.quickOpenService.open({
             onType: async (lookFor, acceptor) => {
                 let error: string | undefined;
+                error = validateInput && lookFor !== undefined ? await validateInput(lookFor) : undefined;
```

Comment:
@ronabet can you clarify why we have the error check initially, which duplicates the check at line `141`?
In addition, I believe it was done purposely to not validate the input on start: https://github.com/eclipse-theia/theia/commit/74d3790981c96ab664df300c9d05fcb31d3a94f8.

Reply:
This would be annoying if there are multiple inputs that need to be entered for creating the artifact and if the user, by mistake, presses enter without giving any value in one of the inputs, then the whole process has to be repeated.
The "validateInput" doesn't get called in unless the user changes the value. Similar code works in pure vscode API.